ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -95,6 +95,18 @@ public void setElevation(@NonNull T view, float elevation) {
     ViewCompat.setElevation(view, PixelUtil.toPixelFromDIP(elevation));
   }
 
+  @Override
+  @ReactProp(
+      name = ViewProps.SHADOW_COLOR,
+      defaultInt = Color.BLACK,
+      customType = "Color")
+  public void setShadowColor(@NonNull T view, int shadowColor) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+      view.setOutlineAmbientShadowColor(shadowColor);
+      view.setOutlineSpotShadowColor(shadowColor);
```

Comment:
Why do we need both?

Reply:
The rendered shadows are a combination of both an ambient shadow (soft) and a spot (hard) shadow which produce the Material design elevation effect.

> In the Material Design environment, virtual lights illuminate the UI. Key lights create sharper, directional shadows, called key shadows. Ambient light appears from all angles to create diffused, soft shadows, called ambient shadows.

![image](https://user-images.githubusercontent.com/6184593/79725792-079e6280-82ea-11ea-8065-49865b3cf533.png)

It's best described here:
https://material.io/design/environment/light-shadows.html#light